nistats/glm.py

```diff
@@ -58,183 +47,394 @@ def data_scaling(Y):
+    Parameters
+    ----------
+    Y : array of shape(n_time_points, n_voxels)
+        the fMRI data
+
+    X : array of shape (n_time_points, n_regressors)
+           the design matrix
 
-    glm_results : dictionary of nistats.regression.RegressionResults instances,
-                 describing results of a GLM fit
+    noise_model : {'ar1', 'ols'}, optional
+        the temporal variance model. Defaults to 'ar1'
 
-    labels : array of shape(n_voxels),
-            labels that associate each voxel with a results key
+    bins : int, optional
+        Maximum number of discrete bins for the AR(1) coef histogram
+
+    Returns
+    -------
+    labels : array of shape (n_voxels),
+        a map of values on voxels used to identify the correseponding the model
+
+    results : dictionary,
+        keys correspond to the different labels values
+        values are RegressionResults instances corresponding to the voxels
     """
+    if noise_model not in ['ar1', 'ols']:
+        raise ValueError(' %s Unknown noise model' % noise_model)
 
-    def __init__(self, X):
-        """
-        Parameters
-        ----------
-        X : array of shape (n_time_points, n_regressors)
-           the design matrix
-        """
-        self.X = X
-        self.labels_ = None
-        self.results_ = None
+    if Y.shape[0] != X.shape[0]:
+        raise ValueError('Response and predictors are inconsistent')
 
-    def fit(self, Y, model='ar1', bins=100):
-        """GLM fitting of a dataset using 'ols' regression or the two-pass
+    # fit the OLS model
+    ols_result = OLSModel(X).fit(Y)
 
-        Parameters
-        ----------
-        Y : array of shape(n_time_points, n_samples)
-            the fMRI data
-        model : {'ar1', 'ols'}, optional
-            the temporal variance model. Defaults to 'ar1'
-        bins : int, optional
-            Maximum number of discrete bins for the AR(1) coef histogram
-        """
-        if model not in ['ar1', 'ols']:
-            raise ValueError('Unknown model')
-
-        if Y.ndim == 1:
-            Y = Y[:, np.newaxis]
-
-        if Y.shape[0] != self.X.shape[0]:
-            raise ValueError('Response and predictors are inconsistent')
-
-        # fit the OLS model
-        ols_result = OLSModel(self.X).fit(Y)
-
-        # compute and discretize the AR1 coefs
-        ar1 = ((ols_result.resid[1:] * ols_result.resid[:-1]).sum(0) /
-               (ols_result.resid ** 2).sum(0))
-        ar1 = (ar1 * bins).astype(np.int) * 1. / bins
-
-        # Fit the AR model acccording to current AR(1) estimates
-        if model == 'ar1':
-            self.results_ = {}
-            self.labels_ = ar1
-            # fit the model
-            for val in np.unique(self.labels_):
-                m = ARModel(self.X, val)
-                self.results_[val] = m.fit(Y[:, self.labels_ == val])
+    # compute and discretize the AR1 coefs
+    ar1 = ((ols_result.resid[1:] * ols_result.resid[:-1]).sum(0) /
```

Comment:
I tend to find it clearer to use the parameter name i.e. axis=0 but maybe that's just me. Same thing for the line below.